src/scmrepo/git/lfs/client.py

```diff
@@ -228,33 +227,48 @@ def __init__(self, url: str, host: str, port: int, path: str):
         super().__init__(url)
         self.host = host
         self.port = port
+        self.username = username
         self.path = path
         self._ssh = _get_ssh_vendor()
 
     @classmethod
     def from_git_url(cls, git_url: str) -> "_SSHLFSClient":
-        result = cls._URL_PATTERN.match(git_url)
-        if not result:
+        if scp_match := SCP_REGEX.match(git_url):
+            # Add an ssh:// prefix and replace the ':' with a '/'.
+            git_url = scp_match.expand("ssh://\1\2/\3")
+
+        parsed = urlparse(git_url)
+        if parsed.scheme != "ssh" or not parsed.hostname:
             raise ValueError(f"Invalid Git SSH URL: {git_url}")
-        host, port, path = result.group("host", "port", "path")
+
+        host = parsed.hostname
+        path = parsed.path.lstrip("/")
```

Comment:
Sorry for an oversight there. I think we should just not add the `.git` in that LFS url instead of removing the suffix here.

Reply:
I don't think that's a robust way to solve it. Omitting the `.git` suffix in `git clone` is a valid SSH URL, but the LFS API URL _must_ use the `.git` suffix. If you prefer, we could _ensure_ it's part of `path` instead:

```suggestion
        path = parsed.path.lstrip("/").removesuffix(".git") + ".git"
```

But IMHO, stripping it from `path` and adding it explicitly where needed/wanted is clearer.

Reply:
Could you point me where it says it has to end with `.git` suffix?

Reply:
Sorry, I read it again and found it.

> By default, Git LFS will append .git/info/lfs to the end of a Git remote url to build the LFS server URL it will use:

Reply:
The Git LFS docs don't state it. See also https://github.com/iterative/scmrepo/pull/346#discussion_r1533632636.

Reply:
@sisp, the doc [here](https://github.com/git-lfs/git-lfs/blame/eda4fd704cf5c17e2c061fb4d435a41e0bc2f83b/docs/api/server-discovery.md#L14-L15) says that it will append `.git/info/lfs`. So, you are right, that we need to add a `.git` to the URL.

> By default, Git LFS will append .git/info/lfs to the end of a Git remote url to build the LFS server URL it will use:

Also, it does equivalent of `path.removesuffix('.git') + '.git'` in the implementation:

https://github.com/git-lfs/git-lfs/blob/eda4fd704cf5c17e2c061fb4d435a41e0bc2f83b/lfsapi/endpoint_finder.go#L225-L229

Reply:
Ah, nice, fantastic research! :bow: Good to know this is platform-dependent behavior but systematic. :+1: